.github/workflows/build-image.yaml

```diff
@@ -16,48 +17,69 @@ jobs:
     name: Build and publish uptime-checker to GHCR
     if: ${{ !github.event.pull_request.head.repo.fork && github.actor != 'dependabot[bot]' }}
     runs-on: ubuntu-latest
-
+    permissions:
+      contents: read
+      packages: write
     env:
       VERSIONED_IMAGE: ghcr.io/getsentry/uptime-checker:${{ github.sha }}
       NIGHTLY_IMAGE: ghcr.io/getsentry/uptime-checker:nightly
+      DOCKERFILE_PATH: Dockerfile
+      PLATFORMS: linux/arm64,linux/amd64
+      IS_MAIN_BRANCH: ${{ github.ref == 'refs/heads/main' }}
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
 
-      - uses: docker/setup-qemu-action@v3
-      - uses: docker/setup-buildx-action@v3
+      - uses: docker/setup-qemu-action@29109295f81e9208d7d86ff1c6c12d2833863392 # v3
+      - uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3
 
       - name: Login to GHCR
         run: docker login --username '${{ github.actor }}' --password '${{ secrets.GITHUB_TOKEN }}' ghcr.io
 
       - name: Build and push to GHCR
-        run: >-
-          docker buildx build
-              --platform linux/arm64,linux/amd64
-               $( [[ "${GITHUB_REF}" == "refs/heads/main" ]] && printf %s "--tag ${NIGHTLY_IMAGE}" )
-              --tag "${VERSIONED_IMAGE}"
-              --file Dockerfile
-              --push
+        shell: bash
+        run: |
+          if [[ "${IS_MAIN_BRANCH}" == "true" ]]; then
+            docker buildx build \
+              --platform "${PLATFORMS}" \
+              --tag "${NIGHTLY_IMAGE}" \
+              --tag "${VERSIONED_IMAGE}" \
+              --file "${DOCKERFILE_PATH}" \
+              --push \
+              .
+          else
+            docker buildx build \
+              --platform "${PLATFORMS}" \
+              --tag "${VERSIONED_IMAGE}" \
+              --file "${DOCKERFILE_PATH}" \
+              --push \
```

Comment:
AFAIK, `--push` should only happen on main branch. But please confirm with the devinfra team.

Reply:
Ah good catch!